### ISSUE TITLE
unnecessarily bound functions

### ISSUE THREAD

**Issue by michaelficarra** [clutchski/coffeelint#15]
Bound functions without any references to their context anywhere in their immediate scope (or their bound children's scope, recursively tested) should produce a warning.

```
=> this # good
=> "this" # bad
=> => => => this # good
=> -> => => this # bad
```
